### PR TITLE
Add basic API documentation.

### PR DESCRIPTION
**Pull request by jankaspar**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Armada API
+
+Armada exposes gRPC and REST api.
+
+## gRPC
+All api is defined in `/pkg/api` folder with `*.proto` files as source for all generated code. 
+
+Folder `/pkg/api` also contains generated clients and together with helper methods from `/pkg/client` provides convenient wayt to call Armada api from go code. See armadactl code for [examples](../cmd/armadactl/cmd/submit.go).
```
Comment (JamesMurkin) [inline docs/api.md:8]:
```suggestion
Folder `/pkg/api` also contains generated clients and together with helper methods from `/pkg/client` provides a convenient way to call Armada api from go code. See armadactl code for [examples](../cmd/armadactl/cmd/submit.go).
```

**Review by JamesMurkin — APPROVED** [review]

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Armada API
+
+Armada exposes gRPC and REST api.
```
Comment (ingvarm-gr) [inline docs/api.md:3]: `api` should be `API` (or possibly `APIs`, if we consider the gRPC and REST endpoints to be separate). Possibly "Armada exposes an API via gRPC or REST"?

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+- [usage.proto](../pkg/api/usage.proto) - methods for reporting of resources usage
+
+## REST
+The REST api only exposes the public part of the gRPC api and it is implemented using grpc-gateway (https://github.com/grpc-ecosystem/grpc-gateway).
+
+Swagger json specification can be found [here](../pkg/api/api.swagger.json) and is also served by armada under `my.armada.deployment/api/swagger.json`
+
+## Authentication
+
+Both gRPC and Rest api support the same set of authentication methods. In the case of gRPC all authentication methods uses `authorization` key in grpc metadata. The REST api use standard http Authorization header (which is translated by grpc-gateway to `authorization` metadata).
```
Comment (ingvarm-gr) [inline docs/api.md:45]: `Rest` should be `REST`.

File: docs/api.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+See helm chart [documentation](./helm/server.md#Authentication) for different server authentication schemes setup.
+
+### No Auth
+For testing armada can be configured to accept no authentication. All operations will use user `anonymous` in this case.
```
Comment (ingvarm-gr) [inline docs/api.md:50]: Probably should have a comma after "testing" ("For tesing, Armada can ...")?

**Review by ingvarm-gr — APPROVED** [review]

**Pull request merged**